qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -131,29 +203,40 @@ def __init__(
             link_parameters: if set to True then the amplitude and DRAG parameters of the
                 X and Y gates will be linked as well as those of the SX and SY gates.
         """
-        super().__init__(basis_gates, default_values)
         self._link_parameters = link_parameters
+        self._use_drag = use_drag
+
+        extra_kwargs = {"use_drag": use_drag, "link_parameters": link_parameters}
+
+        super().__init__(basis_gates, default_values, **extra_kwargs)
+
+    @property
+    def __supported_gates__(self) -> Dict[str, int]:
+        """The gates that this library supports."""
+        return {"x": 1, "y": 1, "sx": 1, "sy": 1}
```

Comment:
It works to leave this as class variable dictionary like it was before (even mypy is okay with it). That is simpler and more readable, but this way makes `__supported_gates__` read-only which might be preferrable.

Reply:
I'm fine with read only since we are creating a library from which to select pulses.